scipy/spatial/transform/_rotation.pyx

```diff
@@ -223,6 +236,140 @@ cdef double[:, :] _compute_euler_from_matrix(
 
     return angles
 
+@cython.boundscheck(False)
+@cython.wraparound(False)
+cdef double[:, :] _compute_euler_from_quat(
+    np.ndarray[double, ndim=2] quat, const uchar[:] seq, bint extrinsic=False
+):
+    # The algorithm assumes extrinsic frame transformations. The algorithm
+    # in the paper is formulated for rotation quaternions, which are stored
+    # directly by Rotation.
+    # Adapt the algorithm for our case by
+    # 1. Replacing "d" with "-d" for intrinsic rotation
+    # 2. Reversing both axis sequence and angles for extrinsic rotations
+    
+    if not extrinsic:
+        seq = seq[::-1]
+
+    cdef int i = _elementary_basis_index(seq[0])
+    cdef int j = _elementary_basis_index(seq[1])
+    cdef int k = _elementary_basis_index(seq[2])
+    
+    # quick renormalization of transformed quaternion, assumes |q| = 1
+    cdef int norm_squared 
```

Comment:
My personal preference is to use `double` type for constants which are semantically real. 

Very minor though.

Reply:
I see! I'll change that right away.
Just out of curiosity, is there any kind of possible performance difference with this?